core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java

```diff
@@ -112,11 +136,14 @@ public void start() {
 
                 final AmbassadorContainer ambassadorContainer = address.getValue();
                 Unreliables.retryUntilSuccess(120, TimeUnit.SECONDS, () -> {
-                    Profiler localProfiler = profiler.startNested("Ambassador container: " + ambassadorContainer.getContainerName());
 
-                    localProfiler.start("Start ambassador container");
+                    AMBASSADOR_CREATION_RATE_LIMITER.doWhenReady(() -> {
+                        Profiler localProfiler = profiler.startNested("Ambassador container: " + ambassadorContainer.getContainerName());
 
-                    ambassadorContainer.start();
+                        localProfiler.start("Start ambassador container");
+
+                        ambassadorContainer.start();
```

Comment:
See #147 and #146 - **must** implement fail-fast so that an unrecoverable failure isn't repeatedly retried. Not yet done.